tests/integration/test_monitor.py

```diff
@@ -0,0 +1,120 @@
+import os
+import tempfile
+
+import numpy as np
+import pytest
+import tensorflow as tf
+
+import gpflow
+from gpflow.utilities.monitor import (
+    ModelToTensorBoardTask,
+    ScalarToTensorBoardTask,
+    TasksCollection,
+    ImageToTensorBoardTask,
+)
+
+
+class Data:
+    num_data = 20
+    num_steps = 10
+    log_dir = tempfile.gettempdir()
+
+
+@pytest.fixture
+def model_and_closure():
+    data = (
+        np.random.randn(Data.num_data, 2),  # [N, 2]
+        np.random.randn(Data.num_data, 2),  # [N, 1]
+    )
+    kernel = gpflow.kernels.SquaredExponential(lengthscales=[1.0, 2.0])
+    model = gpflow.models.GPR(data, kernel, noise_variance=0.01)
+
+    @tf.function
+    def closure():
+        return -1.0 * model.log_likelihood()
+
+    return model, closure
+
+
+def _get_size_directory(dir):
+    """ Calculating the size of a directory (in Bytes) """
+    return sum(d.stat().st_size for d in os.scandir(dir) if d.is_file())
+
+
+def test_logdir_created(model_and_closure):
+    model, closure = model_and_closure
+
+    def elbo_callback():
+        return model.log_likelihood()
+
+    tasks = TasksCollection(
+        [
+            ModelToTensorBoardTask(Data.log_dir, model),
+            ScalarToTensorBoardTask(Data.log_dir, elbo_callback, "elbo"),
+        ]
+    )
+
+    # check existence
+    assert os.path.exists(Data.log_dir) and os.path.isdir(Data.log_dir)
+    size_before = _get_size_directory(Data.log_dir)
+    assert size_before > 0
+
+    opt = tf.optimizers.Adam()
+    for step in range(Data.num_steps):
+        opt.minimize(closure, model.trainable_variables)
+        tasks(step)
+
+    size_after = _get_size_directory(Data.log_dir)
+    assert size_after > size_before
+
+
+def test_ImageToTensorBoardTask():
```

Comment:
Is there some way we could test the output of this (and the following) tests?
If that's too complex, that's okay, but it might be helpful to make clear in the test name that these are smoke tests, not functionality tests...

Reply:
This would cost me more time and don't really see the benefit right now. I check them manually and they look fine.

Reply:
I'll indicate that the tests are smoke tests though.